### PR TITLE
Initial IsolatedStorage tests

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Implements initial test project for IsolatedStorage.
Fixes a scoping problem with BinaryWriter usage.
Need GetFolderPathCore implemented in System.Enviroment for Win32 to write more tests.

### CODE REVIEW

File: src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs [hunk 1]
```diff
@@ -139,10 +139,10 @@ internal static string GetNormalizedStrongNameHash(AssemblyName name)
                     bw.Write(publicKey);
                     bw.Write(name.Version.Major);
                     bw.Write(name.Name);
-                }
 
-                ms.Position = 0;
-                return GetStrongHashSuitableForObjectName(ms);
```
Comment (stephentoub) [inline src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs]: Why this change?  Does BinaryWriter implicitly flush to the underlying stream on every write?

File: src/System.IO.IsolatedStorage/tests/Resources/Strings.resx [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  <data name="IsolatedStorage_OnlyIncreaseUserApplicationStore" xml:space="preserve">
+    <value>Increasing the quota of this scope is not supported.  Only the user application scope’s quota can be increased.</value>
+  </data>
+  <data name="IsolatedStorage_Scope_Invalid" xml:space="preserve">
+    <value>Invalid scope.</value>
+  </data>
+  <data name="IsolatedStorage_Scope_U_R_M" xml:space="preserve">
+    <value>Invalid scope, expected User, User|Roaming or Machine.</value>
+  </data>
+</root>
```
Comment (stephentoub) [inline src/System.IO.IsolatedStorage/tests/Resources/Strings.resx]: These are all necessary due to including the src helper files?

File: src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs (hunk 1 above)
Comment (JeremyKuhne) [inline src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs]: It closes the contained stream's handle by default.

**Comment by stephentoub** [comment]
I still cringe at the practice of including src files into the test compilation, though I realize we do it elsewhere. cc: @weshaggard 

Other than that and a few nit comments/questions, LGTM.  Thanks for adding the test project.

File: src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs]: I see.  And BinaryWriter auto-flushes to the underlying stream on every write (or doesn't have a buffer at all) so you don't need to add flush calls before reading from the MemoryStream?

File: src/System.IO.IsolatedStorage/tests/Resources/Strings.resx (hunk 2 above)
Comment (JeremyKuhne) [inline src/System.IO.IsolatedStorage/tests/Resources/Strings.resx]: I could filter them out to the ones that matter, but I figured it wasn't worth it for a test assembly. The way we generate resources I couldn't link to the other either.

File: src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs (hunk 1 above)
Comment (JeremyKuhne) [inline src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs]: Yes

File: src/System.IO.IsolatedStorage/tests/Resources/Strings.resx (hunk 2 above)
Comment (stephentoub) [inline src/System.IO.IsolatedStorage/tests/Resources/Strings.resx]: I think it'd be better to trim it down to only what's needed.  There's already going to be drift between the two copies as we add new values to the src or update values there, with no reason to also add it to the tests, and it'd be good to keep the duplication to the minimum possible so that we can more easily understand which values are actually needed.  Ideally we wouldn't need this at all.

File: src/System.IO.IsolatedStorage/tests/System.IO.IsolatedStorage.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  <!-- Default configurations to help VS understand the options -->
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Unix_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Unix_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap101_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap101_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net463_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net463_Release|AnyCPU'" />
+  <ItemGroup Condition="'$(PackageTargetFramework)' == ''">
```
Comment (weshaggard) [inline src/System.IO.IsolatedStorage/tests/System.IO.IsolatedStorage.Tests.csproj]: You shouldn't need these in test projects. They are only for producing packages.

Comment (JeremyKuhne) [inline src/System.IO.IsolatedStorage/tests/System.IO.IsolatedStorage.Tests.csproj]: PackageDestination you mean?

Comment (weshaggard) [inline src/System.IO.IsolatedStorage/tests/System.IO.IsolatedStorage.Tests.csproj]: Yes

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Release Build and Test

**Comment by joperezr** [comment]
@JeremyKuhne what's the state of this PR? I would like to merge/close all PRs against dev/api soon and this is one of the last remaining ones.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test this please

**Comment by JeremyKuhne** [comment]
@joperezr Kicking it against the CI again, when it is green I can merge it

**Comment by JeremyKuhne** [comment]
Moving this to master as the CI builds don't appear to be up.
